examples/example_differential_evo_regression.py

```diff
@@ -28,6 +29,16 @@
 
 import cgp
 
+# %%
```

Comment:
mandatory comment on [docopt](http://docopt.org/) being much cooler than argparse ;)

Reply:
Yes, I agree, `docopt` is much nicer and shorter. However, strangely, `docopt` itself is basically not maintained anymore. See this issue: https://github.com/docopt/docopt/issues/371 with some strange replies by the maintainer (?) who "just doesn't have the time to reject all the PRs" ...

Therefore I would prefer to use [`docopt-ng`](https://github.com/bazaar-projects/docopt-ng), a maintained fork.

Two things are suboptimal in my opinion:
1. docopt does not infer the types of arguments? That is why I need to explicitly convert the value from `str` to `int`. Seems strange that they don't offer that.
2. They also just take over the argument as is, i.e. `--max_generations` is not turned into `max_generations` (what `argparse` is offering). That is also a bit strange.

Perhaps I am just not understanding how to it, though. Do you have an idea?

Reply:
> Two things are suboptimal in my opinion:
> 
>     1. docopt does not infer the types of arguments? That is why I need to explicitly convert the value from `str` to `int`. Seems strange that they don't offer that.
> 
>     2. They also just take over the argument as is, i.e. `--max_generations` is not turned into `max_generations` (what `argparse` is offering). That is also a bit strange.
> 
> 
> Perhaps I am just not understanding how to it, though. Do you have an idea?

i think you are right in both cases. you need to manually convert the arguments from str and they keep their name from the docstring. i would argue that both are not an issue tough.

Reply:
Yeah, not an issue, but quite ugly. I also don't understand, why docopt doesn't infer the type from the default value (or optional documentation).

Reply:
This used to be the case, but was then silently removed: https://www.google.com/search?hl=en&q=docopt%20infer%20type

Reply:
Anyway, it is what it is.